### PR TITLE
Add bonemeal and bone

### PR DESCRIPTION
**Pull request by lnjX**
- [x] Add items
- [x] Add function for growing flowers, grass and waterlilys
- [x] Add way to get bones
- [ ] Add function for sapling growing
- [ ] Add function for crops growing
- [x] Improve textures (maybe)

Maybe we can merge this before we have all of them checked. :smile_cat: 

Screenshot:
screenshot_20160305_062835

### CODE REVIEW

**Comment by jbruechert** [comment]
Maybe move this into the bones mod if you are not planning to move the whole bones mod into default.
But nice work!

I think i will try to optimize the bonemeal texture a bit. Now it looks like grey dye.

**Comment by lnjX** [comment]
I think the bones mod is about something quiet different than the bone and bonemeal.
Bones and bonemeal are items.
Bones from the bones mod are only to protect your items, so other players don't just steal them.

**Comment by jbruechert** [comment]
Ok that depends on the way to get them.
By the way here is my changed Bonemeal texture:
default_bonemeal
compared with the old one:
default_bonemeal_old

**Comment by lnjX** [comment]
Yeah I think you should get them by killing monsters (zombie, ghost)
And nice texture :smiley:!

**Comment by lnjX** [comment]
@JBBgameich Would you agree to merge this? Please read all the code carefully and make suggestions (you could use linenotes)!
I think we can merge this, although you can't grow saplings and crops faster with that (we can do this after the rewrite).

File: mods/default/crafting.lua [hunk 1]
```diff
@@ -722,3 +722,10 @@ minetest.register_craft({
 		{"default:stick", "default:stick", "default:stick"},
 	}
 })
+
+minetest.register_craft({
+	output = "default:bonemeal 4",
```
Comment (lnjX) [inline mods/default/crafting.lua:727]: Are 4 enough, too much or good?

File: mods/default/items/bonemeal.lua [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		
+		-- give minetest the changed itemstack
+		return itemstack
+	end
+end
+
+core.register_craftitem("default:bonemeal", {
+	description = "Bone Meal",
+	inventory_image = "default_bonemeal.png",
+	groups = {dye = 1, basecolor_white = 1, excolor_white = 1, unicolor_white = 1},
```
Comment (lnjX) [inline mods/default/items/bonemeal.lua:110]: With this groups you can use it as dye too! Just testing these line notes...

**Comment by jbruechert** [comment]
ok, I think we can merge that

**Pull request merged**